.github/workflows/lint.yml

```diff
@@ -9,37 +9,102 @@ env:
   CARGO_INCREMENTAL: 0
 
 jobs:
-  lint:
-    name: Lint
+  # Runs the underlying job (“workload”) on a self-hosted runner if available,
+  # with the help of a `runner-select` job and a `runner-timeout` job.
+  runner-select:
+    runs-on: ubuntu-22.04
+    outputs:
+      unique-id: ${{ steps.select.outputs.unique-id }}
+      selected-runner-label: ${{ steps.select.outputs.selected-runner-label }}
+      runner-type-label: ${{ steps.select.outputs.runner-type-label }}
+      is-self-hosted: ${{ steps.select.outputs.is-self-hosted }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: '.github'
+      - name: Runner select
+        id: select
+        uses: ./.github/actions/runner-select
+        with:
+          GITHUB_TOKEN: ${{ github.token }}
```

Comment:
I believe this token has way more permissions than it needs, but I think this is a general issue we have, so perhaps we can solve it a different PR.